src/rgw/rgw_http_client.cc

```diff
@@ -367,6 +372,11 @@ int RGWHTTPClient::init_request(const char *method, const char *url, rgw_http_re
   return 0;
 }
 
+int RGWHTTPClient::reset_request()
+{
+  req_data = NULL;
```

Comment:
@dongbula  is this the correct way to reset it? don't we leak req_data here?

Reply:
@yehudasa hi, this reset_request is not responsible for freeing req_data, because reset_request can't judge whether req_data has been freed or not. See RGWHTTPManager::_complete_request, you will find req_data being freed outside HTTPClient class.